src/Common/test/Certificates.Test/ConfigureCertificateOptionsTest.cs

```diff
@@ -17,176 +18,243 @@ public sealed class ConfigureCertificateOptionsTest
-        configurationBuilder.AddCertificate(CertificateName, certificate1FilePath, privateKey1FilePath);
+        configurationBuilder.AddCertificate(certificateName, certificate1FilePath, privateKey1FilePath);
         IConfigurationRoot configurationRoot = configurationBuilder.Build();
 
         IServiceCollection services = new ServiceCollection();
         services.AddLogging();
         services.AddSingleton<IConfiguration>(configurationRoot);
-        services.ConfigureCertificateOptions(CertificateName);
+        services.ConfigureCertificateOptions(certificateName);
 
         await using ServiceProvider serviceProvider = services.BuildServiceProvider(true);
 
         var optionsMonitor = serviceProvider.GetRequiredService<IOptionsMonitor<CertificateOptions>>();
-        optionsMonitor.Get(CertificateName).Certificate.Should().BeEquivalentTo(firstX509);
+        optionsMonitor.Get(certificateName).Certificate.Should().BeEquivalentTo(firstX509);
 
         IOptionsChangeTokenSource<CertificateOptions>[] tokenSources = [.. serviceProvider.GetServices<IOptionsChangeTokenSource<CertificateOptions>>()];
+
         tokenSources.OfType<FilePathInOptionsChangeTokenSource<CertificateOptions>>().Should().HaveCount(2);
         IChangeToken changeToken = tokenSources.OfType<ConfigurationChangeTokenSource<CertificateOptions>>().Single().GetChangeToken();
 
         bool changeCalled = false;
-        _ = changeToken.RegisterChangeCallback(_ => changeCalled = true, "state");
-        configurationRoot[$"{CertificateOptions.ConfigurationKeyPrefix}:{CertificateName}:CertificateFilePath"] = certificate2FilePath;
-        configurationRoot[$"{CertificateOptions.ConfigurationKeyPrefix}:{CertificateName}:PrivateKeyFilePath"] = privateKey2FilePath;
-        configurationRoot.Reload();
-        optionsMonitor.Get(CertificateName).Certificate.Should().BeEquivalentTo(secondX509);
-        changeCalled.Should().BeTrue("file path information changed");
-
-        _ = changeToken.RegisterChangeCallback(_ => changeCalled = true, "state");
-        await File.WriteAllTextAsync(certificate2FilePath, instance1Certificate, TestContext.Current.CancellationToken);
-        await File.WriteAllTextAsync(privateKey2FilePath, instance1PrivateKey, TestContext.Current.CancellationToken);
-        await Task.Delay(2.Seconds(), TestContext.Current.CancellationToken);
-        optionsMonitor.Get(CertificateName).Certificate.Should().BeEquivalentTo(firstX509);
-        changeCalled.Should().BeTrue("file contents changed");
+
+        using (changeToken.RegisterChangeCallback(_ => changeCalled = true, "changed-path"))
+        {
+            changeCalled.Should().BeFalse("nothing has changed yet");
+            configurationRoot[$"{GetConfigurationKey(certificateName, "CertificateFilePath")}"] = certificate2FilePath;
+            configurationRoot[$"{GetConfigurationKey(certificateName, "PrivateKeyFilePath")}"] = privateKey2FilePath;
+            configurationRoot.Reload();
```

Comment:
This is a pretty extreme action that doesn't accurately reflect what happens in an app. It would be better to write to an `appsettings.json` on disk. The system detects the file changes, a sequence of triggers fire, ultimately resulting in a changed value in the options monitor. We should assert that the options monitor value changed, not the underlying triggers, which is an implementation detail that doesn't really matter for what this test is trying to prove.